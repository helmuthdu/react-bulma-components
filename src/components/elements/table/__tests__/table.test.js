import React from 'react';
import { render } from 'react-testing-library';
import { Table } from '..';

describe('Table component', () => {
  it('should render', () => {
    const { asFragment } = render(<Table />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should concat classname in props with classname', () => {
    const { asFragment } = render(
      <Table className="other-class this-is-a-test">
        <tbody>
          <tr>
            <td>Default</td>
          </tr>
        </tbody>
      </Table>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should use inline styles', () => {
    const { asFragment } = render(
      <Table style={{ height: 250 }}>
        <tbody>
          <tr>
            <td>Default</td>
          </tr>
        </tbody>
      </Table>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
