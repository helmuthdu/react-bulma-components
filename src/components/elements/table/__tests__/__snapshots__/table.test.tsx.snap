// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <table
    class="table other-class this-is-a-test"
  >
    <tbody>
      <tr>
        <td>
          Default
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Table component should render 1`] = `
<DocumentFragment>
  <table
    class="table"
  />
</DocumentFragment>
`;

exports[`Table component should use inline styles 1`] = `
<DocumentFragment>
  <table
    class="table"
    style="height: 250px;"
  >
    <tbody>
      <tr>
        <td>
          Default
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
